src/c++/perf_analyzer/concurrency_manager.cc

```diff
@@ -70,14 +70,16 @@ ConcurrencyManager::ConcurrencyManager(
           shared_memory_type, output_shm_size, start_sequence_id,
           sequence_id_range, parser, factory),
       execute_(true), max_concurrency_(max_concurrency)
+{
+  threads_config_.reserve(max_threads);
+}
+
+void
+ConcurrencyManager::InitManagerFinalize()
 {
   if (on_sequence_model_) {
```

Comment:
This same check is in both child classes. Can you move it to the parent class?

Reply:
It is true that right now, both overridden use cases of `InitManagerFinalize()` happen to guard all of their logic in an if conditioned on `on_sequence_model_`, but `InitManagerFinalize()` is not intended to be run only for sequence models, as defined in its description. Therefore, I would prefer to leave these as is.

Reply:
works for me